src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReaderProvider.cs

```diff
@@ -266,7 +266,7 @@ public unsafe MetadataReader GetMetadataReader(MetadataReaderOptions options = M
                 }
 
                 AbstractMemoryBlock metadata = GetMetadataBlock();
-                var newReader = new MetadataReader(metadata.Pointer, metadata.Size, options, utf8Decoder);
+                var newReader = new MetadataReader(metadata.Pointer, metadata.Size, options, utf8Decoder, memoryOwner: this);
```

Comment:
Should PEReader.GetMetadataReader do the same?

Reply:
Yup. Good catch.